### PR TITLE
merchant customers with pending invoices

### LINKED ISSUE #17: Merchant customers with pending invoices
GET /api/v1/merchants/:id/customers_with_pending_invoices

### PR DESCRIPTION
**Pull request by Dpalazzari**
@tmikeschu 

In this PR:

- Add customers_with_pending_invoices method to merchant model
- Add test for pending_invoices method to the merchant model test
- Add customers_with_pending_invoices API endpoint
- Add customers_with_pending_invoices API endpoint test
- Adds some minor refactoring
- ALL TESTS PASS
- ALL SPEC HARNESS TESTS PASS

Closes #17 

FYI @sallymacnicholas

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -2,66 +2,67 @@
   namespace :api, defaults: {format: :json} do
```
Comment (tmikeschu) [inline config/routes.rb:2]: LOOOVE this extra effort. 🚀

File: app/models/merchant.rb [hunk 2]
```diff
@@ -28,6 +28,28 @@ def self.with_most_items(quantity)
       .limit(quantity)
   end
 
+  def customers_with_pending_invoices
```
Comment (tmikeschu) [inline app/models/merchant.rb:31]: The BEAST. Pleasure working with you on this.

File: app/models/merchant.rb [hunk 3]
```diff
@@ -28,6 +28,28 @@ def self.with_most_items(quantity)
...
+    Customer.find_by_sql("select customers.* from customers
+      join invoices
+      on customers.id = invoices.customer_id
+      where invoices.id in (
+        select invoices.id
+        from invoices
+        join transactions
+        on invoices.id = transactions.invoice_id
```
Comment (tmikeschu) [inline app/models/merchant.rb:39]: I have a feeling we could extract this subquery to invoices, if we ever want to.

**Review by tmikeschu — COMMENTED** [review]
Hooray! 🙌 Nice work on this project. You were a great partner, with a great work ethic, and extremely fastidious code review eyes. Thanks for taking our project to the next level. @Dpalazzari

File: app/models/merchant.rb (hunk 2 above)
Comment (Dpalazzari) [reply app/models/merchant.rb:31]: The pleasure was all mine!

File: app/models/merchant.rb (hunk 3 above)
Comment (Dpalazzari) [reply app/models/merchant.rb:39]: someday...

**Comment by Dpalazzari** [comment]
Was a pleasure to work with you @tmikeschu You are a great coder. I have no doubt you will be successful as a programmer in your life after Turing! 👍 🚀 🥇 🎉 🎈

**Pull request merged**